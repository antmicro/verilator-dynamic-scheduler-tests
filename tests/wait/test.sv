module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;
   int value = 0;

   initial begin
     $write("Starting with value == 0\n");
     #2;
     $write("Assigning 1 to value.\n");
     value = 1;
     #1;
     $write("Assigning 2 to value.\n");
     value = 2;
     #1;
     $write("Assigning 0 to value.\n");
     value = 0;
     #1;
     $write("Assigning 2 to value.\n");
     value = 2;
   end

   initial begin
     #1;
     $write("Waiting for value == 2...\n");
     wait(value == 2);
     $write("Waited for value == 2.\n");
     $write("Waiting for value < 2...\n");
     wait(value < 2);
     $write("Waited for value < 2.\n");
     $write("Waiting for value == 0...\n");
     wait(value == 0);
     $write("Waited for value == 0.\n");
     $write("Waiting for 1 < value < 3...\n");
     wait(value > 1 && value < 3);
     $write("Waited for 1 < value < 3.\n");
     $write("*-* All Finished *-*\n");
     $finish;
   end
endmodule
