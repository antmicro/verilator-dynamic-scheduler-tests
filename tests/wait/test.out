Starting with value == 0
Waiting for value == 2...
Assigning 1 to value.
Assigning 2 to value.
Waited for value == 2.
Waiting for value < 2...
Assigning 0 to value.
Waited for value < 2.
Waiting for value == 0...
Waited for value == 0.
Waiting for 1 < value < 3...
Assigning 2 to value.
Waited for 1 < value < 3.
*-* All Finished *-*
